src/v/storage/disk_log_impl.h

```diff
@@ -442,6 +451,30 @@ class disk_log_impl final : public log {
     std::optional<model::offset> _last_compaction_window_start_offset;
 
     size_t _reclaimable_size_bytes{0};
+
+    uint64_t _dirty_segment_bytes{0};
+    uint64_t _closed_segment_bytes{0};
+
+    struct add_tag {};
+    struct subtract_tag {};
+
+    // Update the number of bytes in dirty segments.
+    //
+    // Dirty segments are closed segments which have not yet been cleanly
+    // compacted- i.e, duplicates for keys in this segment _could_ be found in
+    // the prefix of the log up to this segment.
+    //
+    // This value can increase AND decrease. It increases
+    // when a segment is closed, and decreases when the segment is marked as
+    // cleanly compacted. For that reason, one of the tags add_tag or
+    // subtract_tag must be used.
+    template<typename tag>
+    void update_dirty_segment_bytes(uint64_t bytes);
+
+    // Update the number of bytes in closed segments.
+    // Unlike _dirty_segment_bytes, this can only increase.
```

Comment:
This is currently true, but maybe it should be decremented during the case of retention enforcement/segment eviction.